src/constraints/emissions_constraints.jl

```diff
@@ -0,0 +1,156 @@
+#
+# Redistributions in binary form must reproduce the above copyright notice, this
+# list of conditions and the following disclaimer in the documentation and/or other
+# materials provided with the distribution.
+#
+# Neither the name of the copyright holder nor the names of its contributors may be
+# used to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+# OF THE POSSIBILITY OF SUCH DAMAGE.
+# *********************************************************************************
+
+function add_emissions_constraints(m,p)
+	if !isnothing(p.s.site.CO2_emissions_reduction_min_pct)
+		@constraint(m, MinEmissionsReductionCon, m[:Lifecycle_Emissions_Lbs_CO2] <= (1-p.s.site.CO2_emissions_reduction_min_pct)*m[:Lifecycle_Emissions_Lbs_CO2_BAU])
+	end
+	if !isnothing(p.s.site.CO2_emissions_reduction_max_pct)
+		@constraint(m, MaxEmissionsReductionCon, m[:Lifecycle_Emissions_Lbs_CO2] >= (1-p.s.site.CO2_emissions_reduction_max_pct)*m[:Lifecycle_Emissions_Lbs_CO2_BAU])
+	end
+end
+
+### Year 1 Emissions Calculations
+function add_yr1_emissions_calcs(m,p)
+	# Components:
+	m[:yr1_emissions_onsite_fuel_lbs_CO2], m[:yr1_emissions_onsite_fuel_lbs_NOx], 
+	m[:yr1_emissions_onsite_fuel_lbs_SO2], m[:yr1_emissions_onsite_fuel_lbs_PM25] = 
+	calc_yr1_emissions_from_onsite_fuel(m,p; tech_array=p.techs.fuel_burning)
+
+	m[:yr1_emissions_from_elec_grid_lbs_CO2], m[:yr1_emissions_from_elec_grid_lbs_NOx], 
+	m[:yr1_emissions_from_elec_grid_lbs_SO2], m[:yr1_emissions_from_elec_grid_lbs_PM25] = 
+	calc_yr1_emissions_from_elec_grid_purchase(m,p)
+	
+	m[:yr1_emissions_offset_from_elec_exports_lbs_CO2], m[:yr1_emissions_offset_from_elec_exports_lbs_NOx], 
+	m[:yr1_emissions_offset_from_elec_exports_lbs_SO2], m[:yr1_emissions_offset_from_elec_exports_lbs_PM25] = 
+	calc_yr1_emissions_offset_from_elec_exports(m,p;tech_array=p.techs.elec)
+
+	# Total:
+	if p.s.site.include_exported_elec_emissions_in_total
+		include_exported_elec_emissions_in_total = 1
+	else
+		include_exported_elec_emissions_in_total = 0
+	end
+	# "net_if_selected" indicates that emissions totals will be net of (subtract) exports IF the user sets "site.include_exported_elec_emissions_in_total" to true
```

Comment:
I thought this comment was confusing so I clarified in https://github.com/NREL/REopt.jl/pull/28/commits/9526eb041a7842fc7dd98a3d9a416558a01016cc

Reply:
Haha agreed, I copied the comment from the API emissions code